### PR TITLE
Kitsune2 Bootstrap Server--Preparation

### PR DESCRIPTION
**Pull request by neonphog**
The complete boostrap server branch is rather large. Breaking out the dependencies needed and some unrelated CI tooling changes so the actual server pull requests can be reviewed with more focus.

- See the follow on [Core #10](https://github.com/holochain/kitsune2/pull/10) for the core implementation code of the server.
- See the follow on [Testing #11](https://github.com/holochain/kitsune2/pull/11) PR for api endpoint tests and general notes about this effort in the PR description.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -12,14 +12,27 @@ base64 = "0.22.1"
 # shallow-clone byte arrays is a solved problem.
 # bytes is the crate that solves it.
 bytes = { version = "1.8.0", features = ["serde"] }
+# boot_srv for cli param parsing
+clap = "4.5.21"
+# boot_srv uses this to make a best-effort to clean up tempfiles on shutdown.
```
Comment (jost-s) [inline Cargo.toml:17]:
```suggestion
# bootstrap_srv uses this to make a best-effort to clean up tempfiles on shutdown.
```

**Review by jost-s — DISMISSED** [review]

File: Cargo.toml [hunk 2]
```diff
@@ -12,14 +12,27 @@ base64 = "0.22.1"
 # shallow-clone byte arrays is a solved problem.
 # bytes is the crate that solves it.
 bytes = { version = "1.8.0", features = ["serde"] }
+# boot_srv for cli param parsing
```
Comment (ThetaSinner) [inline Cargo.toml:15]:
```suggestion
# bootstrap_srv for cli param parsing
```

Comment (ThetaSinner) [reply Cargo.toml:15]: and a couple more following this

**Review by ThetaSinner — APPROVED** [review]

File: Cargo.toml [hunk 3]
```diff
@@ -12,14 +12,27 @@ base64 = "0.22.1"
...
+clap = "4.5.21"
+# bootstrap_srv uses this to make a best-effort to clean up tempfiles on shutdown.
```
Comment (mattyg) [inline Cargo.toml:17]: :tada:

**Review by ThetaSinner — APPROVED** [review]

**Pull request merged**